pkg/datapath/loader/loader.go

```diff
@@ -202,6 +203,26 @@ func (l *Loader) reloadHostDatapath(ctx context.Context, ep datapath.Endpoint, o
 	objPaths[0], objPaths[1] = objPath, objPath
 	interfaceNames[0], interfaceNames[1] = ep.InterfaceName(), ep.InterfaceName()
 
+	// In Flannel's case or when using ipvlan, HOST_DEV2 is equal to HOST_DEV1
+	// in init.sh and we don't need to attach the same BPF program to the same
+	// interface twice.
+	if option.Config.DatapathMode != datapathOption.DatapathModeIpvlan &&
+		!option.Config.IsFlannelMasterDeviceSet() {
```

Comment:
nit: I guess logically this is something like `if DatapathHasMultipleMasterDevices(option.Config)` so could be factored out into a helper function. But code is also fine as-is.